### ISSUE TITLE
Ease flow dependancy declaration

### ISSUE THREAD

**Issue by Ben8t** [kestra-io/kestra#4703]
### Feature description

One common pattern in Kestra is declarating flow dependancies through Flow Trigger.

One pain arise when user have many upstream dependancies to declare. For example, the following flow depends of three upstream flow - listening to SUCCESS state.

```yaml
id: flow_downstream
namespace: company.team

tasks:
  - id: task_c
    type: io.kestra.plugin.core.debug.Return
    format: "{{ task.id }}"

triggers:
  - id: multiple_listen_flow
    type: io.kestra.plugin.core.trigger.Flow
    conditions:
      - type: io.kestra.plugin.core.condition.ExecutionStatusCondition
        in:
          - SUCCESS
      - id: multiple
        type: io.kestra.plugin.core.condition.MultipleCondition
        window: P1D
        windowAdvance: P0D
        conditions:
          flow_a:
            type: io.kestra.plugin.core.condition.ExecutionFlowCondition
            namespace: company.team
            flowId: flow_a
          flow_b:
            type: io.kestra.plugin.core.condition.ExecutionFlowCondition
            namespace: company.team
            flowId: flow_b
          flow_c:
            type: io.kestra.plugin.core.condition.ExecutionFlowCondition
            namespace: company.team
            flowId: flow_c
```

Here the user have to write more than 20 lines of YAML - where it would be possible to write something like (raw idea)

```yaml
upstreamDependancies:
    namespace: company.team
    flows: [flow_a, flow_b, flow_c]
condition: SUCESS
```

TL;DR; most of the time users need to trigger a flow upon upstream dependancies being in SUCCESS state. This behavior should be less verbose.

pluginDefault makes things so easy to avoid duplicate/boilerplate code - it could be streamlined the same way for creating dependancies between flows :)

**Comment by anna-geller** [comment]
funny you mentioned it now as just this morning I opened an issue for Flow trigger redesign :) https://github.com/kestra-io/kestra-ee/issues/1543

**Comment by anna-geller** [comment]
This will be possible in 0.20 (naming can change as we still iterate on it):

```yaml
id: myflow
namespace: company.team

triggers:
  - id: wait_for_upstream
    type: io.kestra.plugin.core.trigger.Flow
    conditions:
      - id: poll_for_flows
        type: io.kestra.plugin.core.condition.ExecutionsCondition
        preconditions:
          - namespace: company.team
            flowId: myflow1
            states: [SUCCESS, WARNING]
          - namespace: company.team
            flowId: myflow2
            states: [SUCCESS]

tasks:
  - id: hello
    type: io.kestra.plugin.core.log.Log
    message: I'm triggered after two flows!
```
